protocol/superc20/liquidity-migration.md

```diff
@@ -0,0 +1,60 @@
+
+## Alternatives Considered
+
+We have considered multiple approaches:
+
+### Wrap
+
+Introduce a new contract that wraps one or multiple legacy tokens into an `ISuperchainERC20`.
+
+### Convert
+
+Create a new `ISuperchainERC20` extension for the legacy tokens that can be exchanged with the legacy representation using the burn and mint rights of `L2StandardBridge` .
+
+### Mirror
+
+Create an `ISuperchainERC20` representation that behaves (with some exceptions) like the legacy "mirrored" token while allowing compatibility with Interop. Users will not realize that these are two distinct tokens, as the two contracts (legacy and Mirror) will act as secondary entry points to each other.
+
+## Tradeoffs and Comments
+
+- The Convert and Wrap approaches require users to migrate to the new representation to support interoperability, while mirror representation requires no additional action from the user.
+- The Mirror approach could lead to some UX confusion, especially in displaying double balances.
+- For Wrap and Convert `SuperchainERC20` should only work with trusted `OptimismMintableERC20` tokens, without hardcoding legacy token addresses into it. For that, we would need to upgrade the legacy `OptimismMintableERC20Factory` to keep track of legacy tokens for each L1 token. However, verifying these legacy tokens, especially those from unofficial factories, is complex and poses security challenges. The Mirror representation does not face this issue.
+- When accepting multiple legacy tokens for the same interoperability representation, the Convert method is a bit cleaner than Wrap. This is because Wrap locks liquidity and differentiates each valid source, whereas Convert doesn't. Since legacy tokens are burned and minted, convert doesn't lock liquidity. Mirror can, in theory, support multiple representations, but we suggest sticking to a single one.
+
+### **Considerations**
+
+- Withdrawals to L1 will remain problematic when introducing interoperability without making changes to the `L1StandardBridge`. Users might attempt to withdraw an amount greater than what the `L1StandardBridge` can handle. In that case, several options can be considered:
+  - We can modify `L2StandardBridge` to conduct liquidity checks by comparing the total amount deposited with the total amount withdrawn and canceling the `withdrawal` if the amount is exceeded.
+  - Bridge UIs could monitor liquidity availability in both `L1StandardBridge` and choose the best path.
+
+## Proposed Solution: Mirror design
```

Comment:
I would like to see a more detailed outline of the mirror design and would want to open up a public discussion so that app teams can provide input to see if the mirror design will break anything critical for any major apps.

Reply:
Absolutely! We just submitted a new PR with a more detailed outline:
https://github.com/ethereum-optimism/design-docs/pull/36